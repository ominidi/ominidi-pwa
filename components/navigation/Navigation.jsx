import React from 'react';
import PropTypes from 'prop-types';

const Navigation = ({ navItems }) => (
  <nav className="navigation">
    <ul className="navigation__menu">
      {
        navItems.map(item => (
          <li
            className="navigation__item"
            key={item.href}
          >
            <a
              href={item.href}
              title={item.title}
              className={`navigation__link ${item.icon}`}
            >
              {item.text}
            </a>
          </li>
        ))
      }
    </ul>
  </nav>
);

Navigation.propTypes = {
  navItems: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string,
      title: PropTypes.string,
      text: PropTypes.string,
    }),
  ),
};

Navigation.defaultProps = {
  navItems: [],
};

export default Navigation;
