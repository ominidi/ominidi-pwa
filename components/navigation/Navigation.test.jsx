import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Navigation from './Navigation';

describe('Navigation', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Navigation />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render an menu item for each item present in the menuItem prop', () => {
    const navItems = [
      {
        href: '/',
        title: 'Manifest',
        text: 'Manifest',
        icon: 'manifest',
      },
      {
        href: '/photos',
        title: 'Photos',
        text: 'Photos',
        icon: 'photos',
      },
    ];

    const wrapper = shallow(<Navigation navItems={navItems} />);

    expect(wrapper.find('.navigation__item').length).toEqual(navItems.length);
    wrapper.find('.navigation__link').forEach((item) => {
      expect(navItems.some(element => element.href === item.prop('href'))).toBe(true);
      expect(navItems.some(element => element.title === item.prop('title'))).toBe(true);
      expect(navItems.some(element => element.text === item.text())).toBe(true);
    });
  });
});
