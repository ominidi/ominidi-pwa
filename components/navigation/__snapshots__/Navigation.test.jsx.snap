// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly 1`] = `
<nav
  className="navigation"
>
  <ul
    className="navigation__menu"
  />
</nav>
`;
